tools/github-promote-release.ts

```diff
@@ -14,39 +15,47 @@ type Options = {
   platform: 'web' | 'android' | 'ios'
 }
 
-const getReleaseIdBody = async ({ deliverinoPrivateKey, owner, repo, platform }: Options) => {
+const getAllPreReleased = async ({ deliverinoPrivateKey, owner, repo, platform }: Options) => {
   const appOctokit = await authenticate({ deliverinoPrivateKey, owner, repo })
 
   const releases: Releases = await appOctokit.rest.repos.listReleases({
     owner,
     repo,
   })
+  const releasesWithTags = releases.data.filter(
+    (release: ReleaseType) => release.tag_name.includes(platform) && release.prerelease,
+  )
+  const filteredTagNames = releasesWithTags.map((release: ReleaseType) => release.tag_name)
 
-  const result = releases.data.find(release => release.tag_name.includes(platform))
-  if (result && result.prerelease) {
-    console.warn('Unset prerelease tag of ', result.tag_name)
-    return result
+  if (releasesWithTags.length > 0) {
+    console.warn('Unset prerelease tags of ', filteredTagNames)
+    return releasesWithTags
   }
 
   console.warn('No release found to unset the prerelease tag for. Latest release may already be non-prerelease')
   return null
 }
 
 const removePreRelease = async ({ deliverinoPrivateKey, owner, repo, platform }: Options) => {
-  const release = await getReleaseIdBody({ deliverinoPrivateKey, owner, repo, platform })
-  if (!release) {
+  const allPreReleases = await getAllPreReleased({ deliverinoPrivateKey, owner, repo, platform })
+  if (!allPreReleases) {
     return null
   }
   const appOctokit = await authenticate({ deliverinoPrivateKey, owner, repo })
-  const result = await appOctokit.rest.repos.updateRelease({
-    owner,
-    repo,
-    release_id: release.id,
-    prerelease: false,
-    make_latest: platform === 'android' ? 'true' : 'false', // We always want android to be the latest release, so a link to the latest github release will go to the apk
-  })
-  console.warn('Http response code of updating the result: ', result.status)
-  return release
+  await Promise.all(
+    allPreReleases.map(async (preRelease: ReleaseType) => {
+      const result = await appOctokit.rest.repos.updateRelease({
+        owner,
+        repo,
+        release_id: preRelease.id,
+        prerelease: false,
+        make_latest: platform === 'android' ? 'true' : 'false', // We always want android to be the latest release, so a link to the latest github release will go to the apk
+      })
+      console.warn('Http response code of updating the result: ', result.status)
+    }),
+  )
+  // Returning the least release with PreRelease tag
+  return allPreReleases[0]
```

Comment:
:x: Hmm... with the current implementation the information is lost whether there was actually a release promoted since there is no way anymore to tell whether the latest release was a prerelease or a production release. We should only return a prerelease here if it was actually the latest release. To work around that, the current `getAllPreReleased` should just return all releases with the filtering (and udpating) done in this function (removePreRelease, should probably be renamed to `removePreReleases`).

Reply:
yes, but you filter beforehand to only include prereleases, meaning production releases are not part of the array anymore. So if the latest release was a production release, we dont actually promote anything, but there might still be prerelease somewhere.

Reply:
We need to discuss this 🙈